### ISSUE TITLE
The description of most IfcElemenTypes is incomplete

### ISSUE THREAD

**Issue by Gorgious56** [IfcOpenShell/IfcOpenShell#4624]
Is there a way to scrape the full description from the docs ? As it stands the description does not offer a lot of feedback for most IfcElementTypes.

This might be caused by how the data was scraped but almost all descriptions for IfcElementTypes contained in https://github.com/IfcOpenShell/IfcOpenShell/blob/v0.7.0/src/blenderbim/blenderbim/bim/schema/entity_descriptions.json#L18 stop after the sentence : `The set of shared information may include:`

eg. with IfcAlarmType the description states : `The distribution control element type IfcAlarmType defines commonly shared information for occurrences of alarms. The set of shared information may include: `

when [the docs](https://standards.buildingsmart.org/IFC/RELEASE/IFC4/ADD2_TC1/HTML/schema/ifcsharedbldgelements/lexical/ifcdoortype.htm) state 
```
The distribution control element type IfcAlarmType defines commonly shared information for occurrences of alarms. The set of shared information may include:

    common properties with shared property sets
    common representations
    common materials
    common composition of elements
    common ports

It is used to define a alarm type specification indicating the specific product information that is common to all occurrences of that product type. The IfcAlarmType may be declared within IfcProject or IfcProjectLibrary using IfcRelDeclares and may be exchanged with or without occurrences of the type. Occurrences of IfcAlarmType are represented by instances of IfcAlarm. Refer to the documentation at IfcAlarm for supported property sets, materials, composition, and ports.

```

**Comment by Moult** [comment]
This is a bit unfortunate - we don't want to scrape everything, and the docs for types are pretty generic in how they start. Maybe we should treat types as a special case? Not sure. Scraping docs isn't fun.

**Comment by aothms** [comment]
We do want in the core IFC docs to have some sort of demarcation between the "definition" of the entity and the remainder of the text with examples or usage scenarios. [Also for bSDD for example.] So why not make that a community effort?

In cases where the first paragraph break is not a good demarcation of where the "definition" ends, we could add an HTML comment to designate where the "definition" should end. Something like: `<!-- end of entity definition -->`. When scraping the docs we first check for this mark, if it doesn't exist we assume first paragraph break is a good demarcation.

cc @atomczak 

Thoughts?

**Comment by Moult** [comment]
That's a great idea! I would actually say that we should enforce it to be the first paragraph. If you can't write a good introduction paragraph to a concept, well ... somebody has to :)

**Comment by aothms** [comment]
> we should enforce it to be the first paragraph

That would also work for me. But is a little bit less flexible in terms of markup. That would mean that lists cannot be part of the introductory paragraph (which may well be a feature and not a bug, I don't know for sure).

**Comment by atomczak** [comment]
Yes! I was trying to come up with a rule for it using keywords or looking at bullet points, colons, but there are many edge cases in IFC. I think the rules can get us to a certain point, but a human review is needed. We should have exactly the same split for the bSDD definitions.

In ISO23386 they have distinction into a Definition and Description. Could we also have the first part of the current Semantic Definition kept as Semantic Definition, but all else treat as separate part: Description? 

@aothms you mentioned HTML comment, but shouldn't we put those markers in source markdown files, similar to ">Notes" or even "## sections"?

**Comment by atomczak** [comment]
Right. Good idea to start with such a marker. I wrote the definition splitter to place the initial markers in all .md files and will create a PR on IFC-dev later today.
